money-core/src/main/scala/com/comcast/money/core/formatters/FormatterChain.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2012 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.comcast.money.core.formatters
+import com.comcast.money.api.SpanId
+import com.typesafe.config.Config
+import scala.collection.JavaConverters._
+
+case class FormatterChain(formatters: Seq[Formatter]) extends Formatter {
+  override def toHttpHeaders(spanId: SpanId, addHeader: (String, String) => Unit): Unit = formatters.foreach {
+    formatter => formatter.toHttpHeaders(spanId, addHeader)
+  }
+
+  override def fromHttpHeaders(getHeader: String => String, log: String => Unit): Option[SpanId] = formatters.toStream.flatMap {
+    formatter => formatter.fromHttpHeaders(getHeader, log)
+  } headOption
+
+  override def fields: Seq[String] = formatters.flatMap {
+    formatter => formatter.fields
+  }
+
+  override def setResponseHeaders(getHeader: String => String, addHeader: (String, String) => Unit): Unit = formatters.foreach {
+    formatter => formatter.setResponseHeaders(getHeader, addHeader)
+  }
```

Comment:
do we want to offer some information as to the total size of headers we are adding?

Reply:
The `fields` member would return the total list of headers that might be added.  The actual headers depend on the SpanId/SpanContext.

Reply:
I meant more that we might make the request breach max header size for lets say apache. wasnt sure if maybe it was worth giving the user some information about the size in bytes that we are adding

Reply:
I don't think that would be a concern.  Even if someone turned on every single propagator I've included I don't think the total size of the headers + values would exceed a few hundred bytes.